measurements/toxicity/toxicity.py

```diff
@@ -0,0 +1,141 @@
+
+Returns:
+    `toxicity`: a list of toxicity scores, one for each sentence in `predictions` (default behavior)
+    `max_toxicity`: the maximum toxicity over all scores (if `aggregation` = `maximum`)
+    `toxicity_ratio`": the percentage of predictions with toxicity >= 0.5 (if `aggregation` = `ratio`)
+
+Examples:
+
+    Example 1 (default behavior):
+        >>> toxicity = evaluate.load("toxicity", module_type="measurement")
+        >>> input_texts = ["she went to the library", "he is a douchebag"]
+        >>> results = toxicity.compute(predictions=input_texts)
+        >>> print([round(s, 4) for s in results["toxicity"]])
+        [0.0002, 0.8564]
+
+    Example 2 (returns ratio of toxic sentences):
+        >>> toxicity = evaluate.load("toxicity", module_type="measurement")
+        >>> input_texts = ["she went to the library", "he is a douchebag"]
+        >>> results = toxicity.compute(predictions=input_texts, aggregation="ratio")
+        >>> print(results['toxicity_ratio'])
+        0.5
+
+    Example 3 (returns the maximum toxicity score):
+
+        >>> toxicity = evaluate.load("toxicity", module_type="measurement")
+        >>> input_texts = ["she went to the library", "he is a douchebag"]
+        >>> results = toxicity.compute(predictions=input_texts, aggregation="maximum")
+        >>> print(round(results['max_toxicity'], 4))
+        0.8564
+
+    Example 4 (uses a custom model):
+
+        >>> toxicity = evaluate.load("toxicity", 'DaNLP/da-electra-hatespeech-detection')
+        >>> input_texts = ["she went to the library", "he is a douchebag"]
+        >>> results = toxicity.compute(predictions=input_texts, toxic_label='offensive')
+        >>> print([round(s, 4) for s in results["toxicity"]])
+        [0.0176, 0.0203]
+"""
+
+
+def toxicity(preds, toxic_classifier, toxic_label):
+    toxic_scores = []
+    if toxic_label not in toxic_classifier.model.config.id2label.values():
+        logger.warning(
+            "The `toxic_label` that you specified is not part of the model labels. Run `model.config.id2label` to see what labels your model outputs."
+        )
+    else:
+        for pred_toxic in toxic_classifier(preds):
+            hate_toxic = [r["score"] for r in pred_toxic if r["label"] == toxic_label][0]
+            toxic_scores.append(hate_toxic)
+    return toxic_scores
+
+
+@evaluate.utils.file_utils.add_start_docstrings(_DESCRIPTION, _KWARGS_DESCRIPTION)
+class Toxicity(evaluate.Measurement):
+    def _info(self):
+        return evaluate.MeasurementInfo(
+            module_type="measurement",
+            description=_DESCRIPTION,
+            citation=_CITATION,
+            inputs_description=_KWARGS_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "predictions": datasets.Value("string", id="sequence"),
+                }
+            ),
+            codebase_urls=[],
+            reference_urls=[],
```

Comment:
no references, code on github we can reference here?

Reply:
Not really, there is just the dataset that the toxicity model was trained on? https://github.com/bvidgen/Dynamically-Generated-Hate-Speech-Dataset

Not sure if that's helpful

Reply:
Is there a reason we wouldn't want to link to the repo corresponding to the dataset we use here? They use Perspective as opposed to the FAIR model for scoring, but the data and general idea is the same.

https://github.com/allenai/real-toxicity-prompts

Reply:
Is there a reason we wouldn't want to link to the RealToxicityPrompts repo? The classifier is different (Perspective vs. FAIR classifier) but it's the same idea, and RealToxicityPrompts is a canonical citation for the toxicity metric in the past few years.

https://github.com/allenai/real-toxicity-prompts/